### ISSUE TITLE
no exception on binding with set bindDN and empty password

### ISSUE THREAD

**Issue by bewue** [ldapts/ldapts#140]
It does not throw an error if bindDN is set and the password is empty.

This is quite dangerous if you want to check login data of users. Is the behavior of ldapts in this case correct?
Or should I check the login data of users with another method?

**Comment by thernstig** [comment]
@bewue could you give a repro of what you do?

**Comment by pano9000** [comment]
I just noticed the same behaviour, while trying to use `ldapts` - and accidentally passing `undefined` as password to the bind method.

```
import { Client } from "ldapts"

const client = Client({
    url: "ldap://our-ldap-url"
})

const bindDn = "cn=exampleUser,dc=example,dc=com";
const password = undefined;

await client.bind(bindDn, password)

console.log("isConnected?", client.isConnected);
client.unbind()
```
I would expect this to throw an `InvalidCredentialsError` error, however it does not do that, but instead "client.isConnected" will be `true`.

However this seems to actually be depending on our particular LDAP configuration.

When I try the same thing with one of those [public LDAP test servers](https://www.forumsys.com/2022/05/10/online-ldap-test-server/), I actually get the following error, when I try to bind with an `undefined` password:
```
UnwillingToPerformError: unauthenticated bind (DN with no password) disallowed Code: 0x35
```

quick example code:
```
import { Client } from "ldapts"

const client = new Client({
    url: "ldap://ldap.forumsys.com",
});

const bindDn = "cn=read-only-admin,dc=example,dc=com"
const password = undefined;
await client.bind(bindDn, password)

console.log("isConnected?", client.isConnected);

client.unbind()
```

After some additional digging, I also found the following article:
https://blog.devolutions.net/2021/03/why-active-directory-ldap-unauthenticated-binds-should-be-disabled-and-how-to-do-it/

So long story short:

We as library consumers need to make sure we check the password != undefined, and ideally should disable unauthenticated binds if possible in our LDAP/AD servers.
